prettyPrinters/rust_types.py

```diff
@@ -35,7 +35,7 @@ class RustType(object):
 
 # Should be synchronized with `RsDebugProcessConfigurationHelper.RUST_STD_TYPES`
 STD_STRING_REGEX = re.compile(r"^(alloc::([a-z_]+::)+)String$")
-STD_STR_REGEX = re.compile(r"^&str$")
+STD_STR_REGEX = re.compile(r"^&?str\*?$")
```

Comment:
Also, I've checked the same code
```rust
fn main() {
    let mut s = String::from("Hello!");
    let x = s.as_mut();
    let y = x as *mut str;

    let a = "123";
    let b = a as *const str;
    println!("")
}
```
for unix lldb and found that we don't handle even `&mut str`
![image](https://user-images.githubusercontent.com/2539310/115840200-14392780-a424-11eb-94b2-ec2323326d6f.png)

Maybe it's worth generalizing this PR to support any reference and pointer types for strings (including "c"-style for msvc lldb)?
Note, I'm not sure about gdb where we don't provide a summary (if I remember correctly, we can't do it) but it should work well at least for LLDB where we provide it as a summary and don't hide actual children

WDYT?

Reply:
That's a good idea indeed, thank you for the suggestion!